### DISCUSSION TITLE
Missing return value checks of several functions

### DISCUSSION THREAD

**Discussion by patrickjchap** [netdata/netdata#15119] [Q&A]
I am currently running an experimental bug-finding tool on Netdata and have several reports relating to missed error-checks of functions. I don't want to overwhelm with bug reports, so I just thought that I would post a few examples from the results first to see if my understanding is correct and to see if these function calls are even considered potentially fallible. If any of these reports do seem worrisome then a separate bug report can be made.

The function `listen_sockets_add()` returns the value `-1` on error, but is never checked when called in any other function (e.g., `bind_to_this()`).

Similarly, `bind_to_this()` also returns `-1` on error, but is ignored when called in `listen_sockets_setup()`.

The function `getnameinfo()` is checked in multiple places for error (e.g., `connection_allowed()` and `accept_socket()`), except in `connect_to_this_ip46()` where it is ignored.

There are more reports, but I don't want to overwhelm with too many reports so I have just included these few just to see if the reasoning is even correct.

**Comment by andrewm4894** [comment]
oh cool, thanks @patrickjchap! - @netdata/agent any thoughts on these findings.
